mapswipe_workers/mapswipe_workers/generate_stats/generate_stats.py

```diff
@@ -0,0 +1,46 @@
+from mapswipe_workers.definitions import logger
+from mapswipe_workers.definitions import DATA_PATH
+
+from mapswipe_workers.generate_stats import project_stats
+from mapswipe_workers.generate_stats import overall_stats
+
+
+def generate_stats(project_id_list, all_projects=False):
+
+    projects_info_filename = f'{DATA_PATH}/api-data/projects/projects_static.csv'
+    projects_df = overall_stats.get_project_static_info(projects_info_filename)
+
+    projects_info_dynamic_filename = f'{DATA_PATH}/api-data/projects/projects_dynamic.csv'
+    projects_dynamic_df = overall_stats.load_project_info_dynamic(projects_info_dynamic_filename)
+
+    if all_projects:
```

Comment:
Is it correct that you either call `generate_stats([some_list])` or `generate_stats(??, True)`? If so, I would suggest a bit of defensive programming at the start of the function, to check that the args make sense, maybe `if project_id_list is not None: assert all_projects == False` etc... This would help catch bugs where a dev calls the function the wrong way.
Or why not write a wrapper around `generate_stats(project_id_list)` like `generate_stats_for_all_projects()` in which you just get the complete id list like you do here on line 17, and then call `generate_stats(the_full_list)`? I think this would make the intent clearer elsewhere in the codebase.